### PR TITLE
[WIP] initial support for building flatpak

### PR DESCRIPTION
**Pull request by abderrahim**
This currently builds a flatpak bundle rather than publishing to a repository.

Depends on https://github.com/yuzu-emu/build-environments/pull/4

### CODE REVIEW

**Comment by chris062689** [comment]
Eventually when this is ready to publish we'll need to publish to a repository hosted on our servers. 

Minor nit looking at it -- yuzu needs to be lowercase. As it's the official spelling "yuzu"

File: .ci/scripts/flatpak/docker.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ccache -s
+
+sed -e s/@DISPLAY_VERSION@/$1/ -e s/@COMPATIBILITY_REPORTING@/${ENABLE_COMPATIBILITY_REPORTING:-"OFF"}/ dist/org.yuzu_emu.Yuzu.yml.in > org.yuzu_emu.Yuzu.yml
+
+flatpak-builder build --ccache --repo /yuzu/repo org.yuzu_emu.Yuzu.yml
+
+ccache -s
+
+# build a bundle
+flatpak build-bundle --runtime-repo=https://flathub.org/repo/flathub.flatpakrepo /yuzu/repo/ org.yuzu_emu.Yuzu.flatpak org.yuzu_emu.Yuzu
```
Comment (ReaperOfSouls1909) [inline .ci/scripts/flatpak/docker.sh]: Change from Yuzu to yuzu no uppercase

**Comment by abderrahim** [comment]
@DarkLordZach I just did it so we can test it works in CI before merging, I'll remove it once everything is confirmed to work.

**Comment by soredake** [comment]
Any progress on this?

**Comment by soredake** [comment]
https://github.com/flathub/org.citra_emu.citra

**Comment by soredake** [comment]
https://flathub.org/apps/details/org.yuzu_emu.yuzu

**Comment by Morph1984** [comment]
Closing as we now have an officially maintained flatpak repository above

**Pull request closed without merging**